kernel/thread.c

```diff
@@ -29,17 +29,60 @@ extern struct _static_thread_data _static_thread_data_list_start[];
 extern struct _static_thread_data _static_thread_data_list_end[];
 
 #ifdef CONFIG_USERSPACE
-/* Each thread gets assigned an index into a permission bitfield */
-static atomic_t thread_index;
+static int thread_count;
 
-static unsigned int thread_index_get(void)
+/*
+ * Fetch an unused thread ID. Returns -1 if all thread IDs are in use
+ */
+static int thread_index_get(void)
```

Comment:
Might want to rename this function to `get_next_thread_index()` or something -- having it as `thread_index_get()` is slightly misleading as it does a bunch of stuff, including modifying global state.

Reply:
done